### PR TITLE
chore: add a test to specify the expectation when moving between access strategies

### PR DESCRIPTION
**Pull request by nordfjord**
Tested against DynamoStore

### CODE REVIEW

File: tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs [hunk 1]
```diff
@@ -495,4 +495,37 @@ type Tests(testOutputHelper) =
...
 
+    [<AutoData(SkipIfRequestedViaEnvironmentVariable="EQUINOX_INTEGRATION_SKIP_COSMOS")>]
+    let ```Can safely evolve AccessStrategy over time`` cartContext = Async.RunSynchronously <| async {
+        let context = createPrimaryContext log 10
+        let unoptimized = Cart.createServiceWithoutOptimization log context
+        let snapshot = Cart.createServiceWithSnapshotStrategy log context
+        let rollingState = Cart.createServiceWithRollingState log context
+        let sku1 = SkuId(Guid.NewGuid())
+        let sku2 = SkuId(Guid.NewGuid())
+        let sku3 = SkuId(Guid.NewGuid())
```
Comment (nordfjord) [inline tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs:506]: I left this in so you could tell me to make it a one liner and ignore my other styling faults.

Comment (bartelink) [reply tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs:506]: There's a Guid.gen, and if there isnt there should be a SkuId.gen ()
(Then we just need a Tuple.init3 and you have a most elegant one liner lol - I worked in a company that had a Marvel of a library where that absolutely would be added no questions asked)

File: tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs [hunk 2]
```diff
@@ -495,4 +495,37 @@ type Tests(testOutputHelper) =
         verifyRequestChargesMax 1
     }
 
+    [<AutoData(SkipIfRequestedViaEnvironmentVariable="EQUINOX_INTEGRATION_SKIP_COSMOS")>]
+    let ```Can safely evolve AccessStrategy over time`` cartContext = Async.RunSynchronously <| async {
```
Comment (bartelink) [inline tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs:499]: I think we have FsCheck3 in da house so it should just work wihtout the RS?

File: tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs [hunk 3]
```diff
@@ -495,4 +495,37 @@ type Tests(testOutputHelper) =
...
+        do! run snapshot sku1 9
+
+        let! unoptimizedState = unoptimized.Read(cartId)
+        let! snapshotState = snapshot.Read(cartId)
+        let! rollingStateState = rollingState.Read(cartId)
+
+        let qty1 = unoptimizedState.items |> Seq.map (fun x -> x.skuId, x.quantity) |> Map.ofSeq
+        let qty2 = snapshotState.items |> Seq.map (fun x -> x.skuId, x.quantity) |> Map.ofSeq
+
+        test <@ qty1 = Map.ofList [ sku1, 9; sku2, 11 ] @>
```
Comment (bartelink) [inline tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs:527]:
```suggestion
        test <@ qty1 = Map [ sku1, 9; sku2, 11 ] @>
```
Map.ofList and Map.ofSeq are both superfluous lib functions ;)

File: tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs [hunk 4]
```diff
@@ -495,4 +495,37 @@ type Tests(testOutputHelper) =
...
+        // It's safe to move to RollingState
+        do! run rollingState sku3 12
+        // It's not safe to move back to Unoptimized after RollingState
+        do! run snapshot sku1 9
+
+        let! unoptimizedState = unoptimized.Read(cartId)
+        let! snapshotState = snapshot.Read(cartId)
+        let! rollingStateState = rollingState.Read(cartId)
+
+        let qty1 = unoptimizedState.items |> Seq.map (fun x -> x.skuId, x.quantity) |> Map.ofSeq
```
Comment (bartelink) [inline tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs:524]:
```suggestion
        let qty1 = Map [ for x in unoptimizedState.items -> x.skuId, x.quantity ]
```

**Comment by bartelink** [comment]
That's a beautiful test, thanks!

File: tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs (hunk 2 above)
Comment (nordfjord) [reply tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs:499]: It failed locally when I tried

Comment (bartelink) [reply tests/Equinox.CosmosStore.Integration/DocumentStoreIntegration.fs:499]: I'll never doubt you again, promise!

**Pull request merged**